Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -479,13 +478,17 @@ bool Digitizer::convertSignalsToADC(const int det, int& signals)
       // The gain factors
       float padgain = calGainFactorDetValue; // * calGainFactorROC->GetValue(col, row); // PLEASE FIX ME when CCDB is ready
       if (padgain <= 0) {
-        const auto msg = Form("Not a valid gain %f, %d %d %d", padgain, det, col, row);
-        LOG(FATAL) << msg;
+        LOG(FATAL) << "Not a valid gain " << padgain
+                   << ", " << det
+                   << ", " << col
+                   << ", " << row;
       }
 
-      for (time = 0; time < nTimeTotal; time++) {
+      for (int tb = 0; tb < nTimeTotal; tb++) {
+        const int key = Digit::calculateKey(det, row, col);
```

Comment:
This should be done outside the timebin loop.

Reply:
Yes, I will change this.